virttest/gluster.py

```diff
@@ -66,104 +69,130 @@ def is_gluster_vol_started(vol_name):
             "Gluster volume start failed for volume; %s" % vol_name)
         cmd = "gluster volume start %s" % vol_name
-        process.system(cmd)
+        if session:
+            session.cmd(cmd)
+        else:
+            process.system(cmd)
         return True
     else:
         return True
 
 
 @error_context.context_aware
-def gluster_vol_stop(vol_name, force=False):
+def gluster_vol_stop(vol_name, force=False, session=None):
     """
     Stop the volume if it is started
     """
     # Check if the volume is started, if then stop
-    if is_gluster_vol_started(vol_name):
+    if is_gluster_vol_started(vol_name, session):
         error_context.context("Gluster volume stop for volume; %s" % vol_name)
         if force:
             cmd = "echo 'y' | gluster volume stop %s force" % vol_name
         else:
             cmd = "echo 'y' | gluster volume stop %s" % vol_name
-        process.run(cmd, ignore_status=False, shell=True)
+        if session:
+            session.cmd(cmd)
+        else:
+            process.run(cmd, ignore_status=False, shell=True)
         return True
     else:
         return True
 
 
 @error_context.context_aware
-def gluster_vol_delete(vol_name):
+def gluster_vol_delete(vol_name, session=None):
     """
     Delete the volume if it is not started
     """
     # Check if the volume is stopped, if then delete
-    if not is_gluster_vol_started(vol_name):
+    if not is_gluster_vol_started(vol_name, session):
         error_context.context("Gluster volume delete; %s" % vol_name)
         cmd = "echo 'y' | gluster volume delete %s" % vol_name
-        process.run(cmd, ignore_status=False, shell=True)
+        if session:
+            session.cmd(cmd)
+        else:
+            process.run(cmd, ignore_status=False, shell=True)
         return True
     else:
         return False
 
 
 @error_context.context_aware
-def is_gluster_vol_avail(vol_name):
+def is_gluster_vol_avail(vol_name, session=None):
     """
     Check if the volume already available
     """
     cmd = "gluster volume info"
     error_context.context("Gluster volume info failed")
-    output = decode_to_text(process.system_output(cmd))
+    if session:
+        output = session.cmd_output(cmd)
+    else:
+        output = decode_to_text(process.system_output(cmd))
     volume_name = re.findall(r'Volume Name: (%s)\n' % vol_name, output)
     if volume_name:
-        return gluster_vol_start(vol_name)
+        return gluster_vol_start(vol_name, session)
 
 
-def gluster_brick_create(brick_path, force=False):
+def gluster_brick_create(brick_path, force=False, session=None):
     """
     Creates brick
     """
-    if os.path.isdir(brick_path) and force:
-        gluster_brick_delete(brick_path)
+    if session:
+        cmd1_str = "session.cmd_status('[ ! -d %s ]' % brick_path)"
+        cmd2_str = "session.cmd('mkdir -p %s' % brick_path)"
+    else:
+        cmd1_str = "os.path.isdir(brick_path)"
+        cmd2_str = "os.mkdir(brick_path)"
```

Comment:
to address this I have addressed using wrapper `def check_isfile()` in https://github.com/avocado-framework/avocado-vt/pull/1928,
can we have similar such wrapper for `check_isdir()`, `remove_dir()`, `make_dir()` ?
It will help us for implementing APIs supporting local/remote host.